# Hugging Face Vision Stream

Stress the agent loop with streamed image rows. The default target is `ethz/food101`, but the same cell can point at a larger public image dataset.

Useful knobs: `HF_VISION_DATASET`, `HF_VISION_CONFIG`, `HF_VISION_SPLIT`, `HF_VISION_IMAGE_FIELD`, `HF_VISION_LABEL_FIELD`, `HF_VISION_ROWS`.

In [1]:
from pathlib import Path

exec(Path("samples/hf_vision_stream_loader.py").read_text(encoding="utf-8"))


dataset: ethz/food101
config: (default)
split: train
rows_streamed: 50
mean_width: 494.7
mean_height: 465.0
top_labels:
- 6: 50


row,label,image
1,6,384x512 RGB
2,6,512x512 RGB
3,6,512x383 RGB
4,6,512x348 RGB
5,6,512x512 RGB
6,6,512x512 RGB
7,6,512x512 RGB
8,6,512x512 RGB


## Follow-up Work

After the stream runs, inspect label imbalance, image-size variance, and whether image decoding or network throughput is the bottleneck. Add a markdown finding before treating the notebook as evidence.

## Finding

Clean run with `HF_VISION_ROWS=50` and `HUGGINGFACE_HUB_TOKEN` sourced from the research-suite env completed successfully.

Notebook Lens reported `status: ok`, `executed_cells: 1`, `total_code_cells: 1`, `cell_id: hfvisn01`, and about 4.38 seconds elapsed.

The previous failure was resolved by changing the default stream target from unnamespaced `food101` to `ethz/food101`.

The sample streamed 50 rows, produced image-size summaries, and rendered an HTML preview table with row, label, and image metadata.

The saved output reported `mean_width: 494.7`, `mean_height: 465.0`, and `top_labels: 6: 50`; the tiny row count is useful for smoke testing but not for label-distribution conclusions.
